build/net/src/lib.rs

```diff
@@ -82,23 +90,65 @@ pub fn load_net_config() -> Result<NetConfig> {
     Ok(cfg)
 }
 
+pub fn generate_port_consts(
+    config: &NetConfig,
+    mut out: impl std::io::Write,
+) -> Result<(), std::io::Error> {
+    let ports = config
+        .vlans
+        .iter()
+        .map(|v| v.port)
+        .collect::<HashSet<_>>()
+        .len();
+    assert!(ports <= 2);
+    writeln!(out, "pub const PORT_COUNT: usize = {};", ports.max(1))
+}
+
 pub fn generate_vlan_consts(
     config: &NetConfig,
     mut out: impl std::io::Write,
 ) -> Result<(), std::io::Error> {
-    let vlan = config.vlan.unwrap();
-    let end = vlan.start + vlan.count;
-    if end > 0xFFF {
-        panic!("Invalid VLAN range (must be < 4096)");
+    if config.vlans.iter().any(|v| v.vid > 0xFFF) {
```

Comment:
Is there code that ensures that the vids are unique (I don't see it)? Is it important that the vids be unique (I'm guessing so)?

Would it be worth representing this as a vid-keyed map in the TOML to make this clear? (I'm not sure.)

Reply:
Done in the Grand VLAN refactoring!